tests/test_openclhpp.cpp

```diff
@@ -4336,4 +4337,52 @@ void testTemplateGetImageRequirementsInfo()
 void testTemplateGetImageRequirementsInfo() {}
 #endif // cl_ext_image_requirements_info
 
+static cl_mem clCreateFromGLBuffer_testgetObjectInfo(cl_context context,
+                                                     cl_mem_flags flags,
+                                                     cl_GLuint bufobj,
+                                                     cl_int *errcode_ret,
+                                                     int num_calls)
+{
+    (void)num_calls;
+
+    TEST_ASSERT_EQUAL_PTR(make_context(0), context);
+    TEST_ASSERT_EQUAL(0, flags);
+    if (errcode_ret)
+        *errcode_ret = CL_SUCCESS;
+    return make_mem(0);
+}
+
+static cl_int clGetGLObjectInfo_testgetObjectInfo(cl_mem memobj,
+                                                  cl_gl_object_type *type,
+                                                  cl_GLuint *gl_object_name,
+                                                  int num)
+{
+    memobj = make_mem(0);
+    *type = CL_GL_OBJECT_TEXTURE2D_ARRAY;
+    *gl_object_name = 0;
+    return CL_SUCCESS;
+}
+
+cl_int clReleaseMemObject_testgetObjectInfo(cl_mem memobj, int ret)
+{
+    return CL_SUCCESS;
+}
+
+void testgetObjectInfo() {
+    cl_mem_flags flags = 0;
+    cl_int err = 0;
+    cl_GLuint bufobj = 0;
+    cl_mem memobj = make_mem(0);
+    cl_gl_object_type type = CL_GL_OBJECT_TEXTURE2D_ARRAY;
+    clGetGLObjectInfo_StubWithCallback(clGetGLObjectInfo_testgetObjectInfo);
+    clCreateFromGLBuffer_StubWithCallback(
+        clCreateFromGLBuffer_testgetObjectInfo);
+    clReleaseMemObject_StubWithCallback(clReleaseMemObject_testgetObjectInfo);
+    cl::BufferGL buffer(contextPool[0], flags, bufobj, &err);
+
+    buffer.getObjectInfo(&type, &bufobj);
```

Comment:
Couple of suggestions to improve coverage here:

1. Please test that the return value for `buffer.getObjectInfo()` is `CL_SUCCESS`, as expected.
2. Please test that `type` and `bufobj` are returned as expected.
3. In order to propertly test (2) we should initialize the values of `type` and `bufobj` to something different, because otherwise this test will pass if this query does nothing.

Reply:
Changed to have 2 iterations

Reply:
Thanks, I don't think the two iterations are necessary though, especially since we are only creating one `cl::BufferGL`.

Reply:
Remove iteration